src/__tests__/controllers/CourseController.test.ts

```diff
@@ -0,0 +1,199 @@
+import { Course } from "../../entities/Course";
+import { CourseController } from "../../controllers/CourseController";
+
+import { getMockReq, getMockRes } from "@jest-mock/express";
+import "regenerator-runtime/runtime";
+import { Repository } from "typeorm";
+import { mock, MockProxy, mockReset } from "jest-mock-extended";
+import { Builder } from "builder-pattern";
+
+let mockCourseRepository: MockProxy<Repository<Course>>;
+
+let courseController: any; // SUT
```

Comment:
Con el type que definiste ya se le puede dar un tipo a esto 🙌

```suggestion
let courseController: TController<Course>; // SUT
```